### PR TITLE
docs: Adds Clay CSS 2 Recipes Getting Started

### PR DESCRIPTION
**Pull request by pat270**
issue #80

Instead of trying to figure out all the questions devs may have about Clay CSS, I thought documenting the framework through a cookbook format might be better answering questions that may arise. I wasn't sure where to put it, let me know if there's a better place and anything that needs more clarification.

### CODE REVIEW

File: css/clay_css_2_recipes/02_buttons.md [hunk 1]
```diff
@@ -0,0 +1,445 @@
+# Buttons
+
+Clay CSS follows Bootstrap 4's base + modifier pattern, similar to [OOCSS](https://github.com/stubbornella/oocss/wiki). The base class, `.btn`, should only contain styles common to all `.btn` elements, such as `border-style`, `border-width`, `display`, `font-size`, and any other `button` related css resets.
+
+The modifier, `.btn-primary`, provides the styles for the main variant of `.btn`, such as `background-color`, `border-color`, and `color`. The modifier, `.btn-secondary`, provides styles for the next `.btn` variant.
+
+This recipe will modify the button classes in Clay CSS to match Material Design's shown in [https://material.io/components/buttons/](https://material.io/components/buttons/). The ripple effect on click requires external javascript and is outside the scope of this framework.
```
Comment (wincent) [inline css/clay_css_2_recipes/02_buttons.md:7]: nit: javascript → JavaScript

File: css/clay_css_2_recipes/02_buttons.md [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+
+.btn-unstyled {
+	letter-spacing: normal;
+	text-transform: none;
+}
+```
+
+## Button Variants
+
+Material Design uses a white overlay and opacity with javascript to create the ripple effect on click. We will try to match the `:active` color as close as possible.
```
Comment (wincent) [inline css/clay_css_2_recipes/02_buttons.md:104]: "JavaScript" here too.

**Comment by matuzalemsteles** [comment]
hey @pat270 it would be great to add that to clayui.com 😁

**Comment by wincent** [comment]
Looks like this needs a `yarn format`. The link checker is also failing to verify those font links (`https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500;700&display=swap`) even though they are valid. It is returning a 400 status, probably sniffing something "bot-like" about the request headers. If we can't tweak the headers, we could just add a blacklist to the link checker to avoid checking this kind of link — the intention isn't that literally every URL in the document should be correct; it's that links that a user might actually click on should lead somewhere (not so much the case for these font links, of which only one of them is even clickable and perhaps it shouldn't even be). I'll look into it and report back.

About where this should go: I'm open to it going here or anywhere. Even existing is already valuable.

**Comment by wincent** [comment]
@pat270: I pushed some commits on top to fix CI (formatting and bad link check). I'm going to merge this because CI is green, so that it doesn't sit in limbo while we decide whether this or clayui.com is the right home for it. (I suspect that getting it into clayui.com will take a bit longer. But if we do move it there, we can link to it there from here easily enough.)

**Comment by pat270** [comment]
Thanks @wincent @matuzalemsteles, we can drop them in the Clay docs site.  It doesn't matter to me where, maybe we can hash out the docs here first? BTW this is for 2.x.

**Comment by matuzalemsteles** [comment]
@pat270 no problem, we can do that.

**Pull request merged**